pkg/controller/controller.go

```diff
@@ -483,6 +486,28 @@ func StartController(ctx context.Context, cfg *config.Config, defaultNamespace s
 		return errors.Wrapf(err, "error building FlyteWorkflow clientset")
 	}
 
+	// Create FlyteWorkflow CRD if it does not exist
+	apiextensionsClient, err := apiextensionsclientset.NewForConfig(kubecfg)
+	if err != nil {
+		return errors.Wrapf(err, "error building apiextensions clientset")
+	}
+
+	_, err = apiextensionsClient.ApiextensionsV1().CustomResourceDefinitions().Get(ctx, flyteworkflow.CRD.Name, v1.GetOptions{})
```

Comment:
I think we may want to put this behind a flag... propeller may not have the permission to interact with CustomResourceDefinitions... we can default the flag to enable this but give people a way out if needed...

Reply:
Does propeller today not need perms to fetch / read crd?

Reply:
I like the flag. We can get rid of the "get" before "create" here then?

```
if cfg.CreateFlyteWorkflowCRD {
    apiextensionsClient, err := ...
    
    _, err := apiextensionsClient. ... .Create(ctx, &flyteworkflow.CRD, ..)
    if err != nil {
        if apierrors.IsAlreadyExists(err) {
           logger.Warnf(ctx, "already exists")
        } else {
            return errors.Wrapf(err, "failed to create FlyteWorfklow CRD")
        }
    }
}
```